### PR TITLE
Mention that drop is not called for statically spawned tasks

### LINKED ISSUE #294: Destructors aren't run for detached tasks
I noticed today that when I `.detach()` a task, its `drop()` is never run.
Is this to be expected?
If yes, this should probably be documented in `Task::detach()` docs...

Using smol v1.3.0.

### PR DESCRIPTION
**Pull request by notgull**
Closes #294

### CODE REVIEW

**Comment by taiki-e** [comment]
This is an issue that can generally be well handled by using a oneshot pattern, right? If so, it would be nice to mention it.

**Comment by notgull** [comment]
> This is an issue that can generally be well handled by using a oneshot pattern, right?

Could you elaborate on what you mean by "oneshot pattern"?

**Comment by taiki-e** [comment]
> "oneshot pattern"

That may not have been the official name, but I meant usage of channels like the first example of https://docs.rs/tokio/latest/tokio/sync/oneshot/index.html#examples.

**Comment by notgull** [comment]
I'm sorry, I'm not sure I understand you. What do you mean by "oneshot pattern", and can we provide an example of using this pattern in the docs?

**Comment by taiki-e** [comment]
I guess the code that had the problem is like the following code with println replaced by drop,

```rs
#[tokio::main]
async fn main() {
  tokio::spawn(async {
    tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
    println!("this may or may not be shown");
  });
}
```

[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=56f5d7c34225bee277fa8d39435e5c3c)

However, you can use oneshot channel (or doing the equivalent with a normal channel) to force the program not to exit until the task is finished.

```rs
#[tokio::main]
async fn main() {
  let (s, r) = tokio::sync::oneshot::channel();
  tokio::spawn(async {
    tokio::time::sleep(tokio::time::Duration::from_secs(1)).await;
    println!("this always be shown");
    let _ = s.send(());
  });
  let _ = r.await;
}
```

[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=a2579f86ce2c9fb40af26462c0ea7f2e)

I personally called such a pattern "oneshot pattern", but I don't know if that was an idiomatic term. I'm sorry for confusing by using it.

(I used tokio in example since smol is unavailable on playground, but the same pattern should be available using single-capacity channel of async-channel.)
